src/main/kotlin/org/openrewrite/kotlin/KotlinTypeSignatureBuilder.kt

```diff
@@ -180,6 +189,23 @@ class KotlinTypeSignatureBuilder(private val firSession: FirSession, private val
         }
     }
 
+    private fun anonymousClassSignature(type: FirAnonymousObject): String {
+        val sig = StringBuilder(type.symbol.classId.asFqNameString())
+        for (supertype in type.superTypeRefs) {
```

Comment:
Note:
AnonymousObjects in Java include a unique ID like $1, $2, etc.

In Kotlin, each anonymous object is named `<anonymous>` in the FIR. So, we append the super-types to the name to distinguish the signature.